src/concurrency/sync.rs

```diff
@@ -116,13 +116,25 @@ struct RwLock {
 
 declare_id!(CondvarId);
 
+#[derive(Debug, Copy, Clone)]
+pub enum RwLockMode {
+    Shared,
+    Exclusive,
```

Comment:
Let's use Rut terminology here, not Windows terminology.
It's called "RwLock". The two modes should be "read" and "write". :)